modules/blocks.js

```diff
@@ -228,12 +230,32 @@ Blocks.prototype.isCleaning = {
  * Handle modules initialization.
  * Modules are not required in this file.
  */
-Blocks.prototype.onBind = function() {
+Blocks.prototype.onBind = function(scope) {
 	// TODO: move here blocks submodules modules load from app.js.
+	modules = {
+		cache: scope.cache,
+	};
+
 	// Set module as loaded
 	__private.loaded = true;
 };
 
+/**
+ * Emit socket notification `blocks/change`.
+ *
+ * @param {Block} block
+ * @todo Add description for the params
+ * @todo Add @returns tag
+ */
+Blocks.prototype.onNewBlock = function(block) {
+	modules.cache.clearBlockRelatedCache(err => {
```

Comment:
There is already two methods `removeByPattern` and `deleteJsonForKey` in general. I would rather suggest to develop a generic method `clearCacheFor(scope)`, scope can be `blocks`, `transactions` or in future any thing. This way we can make that single method useable on different cases.